src/internal_events/statsd.rs

```diff
@@ -93,3 +94,29 @@ impl<T: std::fmt::Debug + std::fmt::Display> InternalEvent for StatsdSocketError
         );
     }
 }
+
+#[derive(Debug)]
+pub struct StatsdInvalidMetric {
+    pub value: MetricValue,
+    pub kind: MetricKind,
```

Comment:
Could these just be references? It'd be nice to avoid the `clone` if it's not too much trouble, but I'm not hugely worried about it since it's an error path.

Reply:
Yes, of course that makes sense. It might be an error path, but there is still the potential for it to be called many times per second.

I wonder though if it would be better for MetricKind to be Copy since it is such a simple struct? (https://github.com/timberio/vector/blob/master/src/event/metric.rs#L21)

Reply:
I have made them both references. It just feels more aesthetic with both fields being references rather than one Copy and the other a reference!